rust_icu_ucal/src/lib.rs

```diff
@@ -217,4 +331,86 @@ mod tests {
         let tz_version = super::get_tz_data_version().expect("get_tz_data_version works");
         assert!(re.is_match(&tz_version));
     }
+
+    #[test]
+    fn test_get_set_millis() -> Result<(), common::Error> {
+        let now = get_now();
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        assert!((now - cal.get_millis()?).abs() < 1000f64);
+
+        let date = now + 17.0;
+        cal.set_millis(date)?;
+        assert_eq!(cal.get_millis()?, date);
+        Ok(())
+    }
+
+    #[test]
+    fn test_set_date() -> Result<(), common::Error> {
+        // Thu May 07 2020 21:00:00 GMT-0400 (US Eastern Daylight Time)
+        let time_a: UDate = 1588899600000.0;
+        // Mon May 04 2020 21:00:00 GMT-0400 (US Eastern Daylight Time)
+        let time_b: UDate = 1588640400000.0;
+
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        cal.set_millis(time_a)?;
+        cal.set_date(2020, 4 /* May */, 4)?;
+        assert_eq!(cal.get_millis()?, time_b);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_set_date_time() -> Result<(), common::Error> {
+        // Thu May 07 2020 21:26:55.898 GMT-0400 (US Eastern Daylight Time)
+        let time_a: UDate = 1588901215898.0;
+        // Mon May 04 2020 21:00:00.898 GMT-0400 (US Eastern Daylight Time)
+        let time_b: UDate = 1588640400898.0;
+
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        cal.set_millis(time_a)?;
+        cal.set_date_time(2020, 4 /* May */, 4, 21, 0, 0)?;
+
+        assert_eq!(cal.get_millis()?, time_b);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_get() -> Result<(), common::Error> {
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+        // Thu May 07 2020 21:26:55.898 GMT-0400 (US Eastern Daylight Time)
+        cal.set_millis(1588901215898.0)?;
+
+        assert_eq!(
+            cal.get(UCalendarDateFields::UCAL_DAY_OF_WEEK)?,
+            UCalendarDaysOfWeek::UCAL_THURSDAY as i32
+        );
+
+        assert_eq!(cal.get(UCalendarDateFields::UCAL_MILLISECOND)?, 898);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_offsets_and_daylight_time() -> Result<(), common::Error> {
+        let mut cal = UCalendar::new("America/New_York", "en-US", UCalendarType::UCAL_GREGORIAN)?;
+
+        let expected_zone_offset: i32 = -5 * 60 * 60 * 1000;
+        let expected_dst_offset: i32 = 1 * 60 * 60 * 1000;
+
+        cal.set_date_time(2020, 4 /* May */, 7, 21, 0, 0)?;
```

Comment:
Nit: would it be worthwhile to define separate enum types for each of these params?  They are kind of easy to mix up at the moment. 0 for January is especially weird.